x-pack/plugins/ml/public/application/model_management/test_models/models/text_expansion/text_expansion_inference.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { i18n } from '@kbn/i18n';
+import { SUPPORTED_PYTORCH_TASKS } from '@kbn/ml-trained-models-utils';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { trainedModelsApiProvider } from '../../../../services/ml_api_service/trained_models';
+import { InferenceBase, INPUT_TYPE, type InferResponse } from '../inference_base';
+import { getTextExpansionOutputComponent } from './text_expansion_output';
+import { getTextExpansionInput } from './text_expansion_input';
+
+export interface TextExpansionPair {
+  token: string;
+  value: number;
+}
+
+export interface FormattedTextExpansionResponse {
+  text: string;
+  score: number;
+  originalTokenWeights: TextExpansionPair[];
+  adjustedTokenWeights: TextExpansionPair[];
+}
+
+export type TextExpansionResponse = InferResponse<
+  FormattedTextExpansionResponse,
+  estypes.MlInferTrainedModelResponse
+>;
+
+export class TextExpansionInference extends InferenceBase<TextExpansionResponse> {
+  protected inferenceType = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
+  protected inferenceTypeLabel = i18n.translate(
+    'xpack.ml.trainedModels.testModelsFlyout.textExpansion.label',
+    { defaultMessage: 'Text expansion' }
+  );
+  protected info = [
+    i18n.translate('xpack.ml.trainedModels.testModelsFlyout.textExpansion.info', {
+      defaultMessage:
+        'Expand your search to include relevant terms in the results that are not present in the query.',
+    }),
+  ];
+
+  public queryText$ = new BehaviorSubject<string>('');
```

Comment:
should it be private? 

```suggestion
  private queryText$ = new BehaviorSubject<string>('');
```

Reply:
Yes, and in `QuestionAnsweringInference` and `ZeroShotClassificationInference`. Good spot.
Updated in https://github.com/elastic/kibana/pull/159150/commits/17c1c0f1e6a1bc4d6fe374496560f0d95f374e04 and a718f2b434013e7a923d26588af2bdebacff1e5d